cpp/include/cudf/io/text/data_chunk_source_factories.hpp

```diff
@@ -136,6 +136,50 @@ class istream_data_chunk_reader : public data_chunk_reader {
   std::vector<host_ticket> _tickets;
 };
 
+/**
+ * @brief a reader which produces views of device memory which contain a copy of the data from a
+ * string
+ *
+ */
+class host_span_data_chunk_reader : public data_chunk_reader {
+ public:
+  host_span_data_chunk_reader(cudf::host_span<const char> data) : _data(data) {}
+
+  void skip_bytes(std::size_t read_size) override
+  {
+    if (read_size > _data.size() - _position) { read_size = _data.size() - _position; }
+    _position += read_size;
+  }
+
+  std::unique_ptr<device_data_chunk> get_next_chunk(std::size_t read_size,
+                                                    rmm::cuda_stream_view stream) override
+  {
+    CUDF_FUNC_RANGE();
+
+    if (read_size > _data.size() - _position) { read_size = _data.size() - _position; }
+
+    // get a device buffer containing read data on the device.
+    auto chunk = rmm::device_uvector<char>(read_size, stream);
+
+    // copy the host data to device
+    CUDF_CUDA_TRY(cudaMemcpyAsync(  //
```

Comment:
While I myself prefer the trailing comments some times, our use of clang-format has been much improved recently, and I've been told circumventing it is unsavory.
```suggestion
    CUDF_CUDA_TRY(cudaMemcpyAsync(
```